### DISCUSSION TITLE
Set margin and padding as detached

### DISCUSSION THREAD

**Discussion by BadPoulet** [GrapesJS/grapesjs#5858] [Q&A] [answered]
I'm having an issue on my setup and on the grapesjs demo. When setting a margin or padding through the Dimension sector in the Style Manager, it sets all other values to zero, which I do not want. I would like to have only the property that I entered value to be saved, i.e. **padding-top: 10px;** rather than **padding: 10px 0 0 0;"

![image](https://github.com/GrapesJS/grapesjs/assets/42976385/56acd450-4d1e-4fb6-a22a-0321f5e143a4)
![image](https://github.com/GrapesJS/grapesjs/assets/42976385/ffaac23b-becb-47eb-9257-e76d181cb90f)
![image](https://github.com/GrapesJS/grapesjs/assets/42976385/fda830f2-cecf-48bd-9422-d5877d5a1da5)

My guess is because both padding and margin are set to **detached=false**, which writes CSS with the shorthand padding and margin.

Where would I go to set this property on the margin and padding properties?

Thanks.

**Comment by Sw33tgt** [comment, marked as answer]
Not sure if it's the right way, but something you could try is changing the property after the style manager has been initiated.
Something like:  `editor.StyleManager.getProperty('dimension','margin').set('detached','true')`
